### PR TITLE
Implement Segregated Witness

### PR DESCRIPTION
**Pull request by jameshilliard**
This is mainly for tracking SegWit progress and testing, it can be tested on segnet.

### CODE REVIEW

**Comment by jtoomim** [comment]
Isn't this project being developed upstream? I do not expect development on it to be faster here than upstream, and I think that discussing this PR here would fracture the conversation. We can pull it from upstream (Core) once it's merged into their master branch, the same as we do all other Core commits that we don't explicitly reject.

Note: This is likely to be another trolling attempt. Keep politics in the appropriate forums. Github is for code.

**Comment by toomim** [comment]
I don't believe in trolls. Just mistakes.

**Comment by Levino** [comment]
@jtoomim Agreed. Should be closed.

**Comment by pygeek** [comment]
@jtoomim @Levino I also agree. Should be closed.

**Comment by ghost** [comment]
I guess the thing that would be worth doing on the Bitcoin Classic side is to refactor SegWit as a hard fork rather than a soft fork.

But that can wait until 2MB is deployed and Pieter has finished coding it for Core.

**Comment by Aquentus** [comment]
Nack for reasons given by jtoomim, but also because it should be implemented as a hardfork rather than as a messy, complicated, soft fork.

**Comment by opetruzel** [comment]
@jameshilliard Has any work on a HF implementation been done, as well? Or, is the soft fork version the only effort thus far? Genuinely curious.

**Comment by jameshilliard** [comment]
@opetruzel There has been work done on both variants although the plan was to do it as a soft fork since it can be deployed safety faster than a hard fork.

File: qa/rpc-tests/test_framework/comptool.py [hunk 1]
```diff
@@ -41,17 +41,32 @@ def wait_until(predicate, attempts=float('inf'), timeout=float('inf')):
 
     return False
 
+class RejectResult(object):
+    '''
+    Outcome that expects rejection of a transaction or block.
+    '''
```
Comment (orangeman) [inline qa/rpc-tests/test_framework/comptool.py:47]: What could possibly go wrong here?
I can't see any risks merging as is.

FULL ACK from my side :+1: 

Bitcoin really needs more transaction throughput to grow!
We need to get a release ready before its too late...

**Comment by orangeman** [comment]
**+1 Deploy as hard fork!**

Why are hard forks more safe?

**Comment by jtoomim** [comment]
This branch has merge conflicts. It also does not seem to be the right forum, as @jameshilliard did not write any of the code in this PR, and is submitting incomplete code, and is doing so in a way that [violates principles for a good work flow](https://github.com/bitcoinclassic/bitcoinclassic/pull/8#issuecomment-172166900). This code can be pulled in from our upstream, Core master, once it is committed there. 

At an absolute minimum, please make sure your code merges properly, compiles, and passes unit tests before opening a pull request.

I am closing this.

**Pull request closed without merging**